### PR TITLE
Fixed #32203 Catalog Widget. Select with product conditions displays…

### LINKED ISSUE #32203: Catalog Widget. Select with product conditions displays incorrect attributes.
### Preconditions (*)

1. Magento 2.4.1
2. Install last version magento/module-page-builder

### Steps to reproduce (*)

1. Go to Admin > Content > Pages > Home page
2. Content -> click button Edit with Page Builder
3. Add Content -> Products -> click Edit
4. Open Conditions

### Expected result (*)

1. In the select in the Product Attributes section, attributes are loaded that have the configuration value is_used_for_promo_rules=0 in the database.
2. Example Dynamic SKU
08288a2105

### Actual result (*)

1. Attributes with the value is_used_for_promo_rules = 0 should not be included in the condition.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [1] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by evgeniykapelko**
… incorrect attributes.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32203

### Manual testing scenarios (*)

1. see magento/magento2#32203

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by evgeniykapelko** [comment]
@magento run all tests

**Comment by evgeniykapelko** [comment]
@magento run all tests

File: app/code/Magento/CatalogWidget/Model/Rule/Condition/Product.php [hunk 1]
```diff
@@ -38,6 +38,13 @@ class Product extends \Magento\Rule\Model\Condition\Product\AbstractProduct
      */
     protected $storeManager;
 
+    /**
+     * Attribute data key that indicates whether it should be used for rules
+     *
+     * @var string
+     */
+    protected $_isUsedForRuleProperty = 'is_used_for_promo_rules';
```
Comment (BarnyShergold) [inline app/code/Magento/CatalogWidget/Model/Rule/Condition/Product.php:46]: This is a static value. You should use a constant for this. For a start it is more efficient.

File: app/code/Magento/CatalogWidget/Model/Rule/Condition/Product.php [hunk 2]
```diff
@@ -97,6 +104,13 @@ function ($attribute) {
 
         $attributes = [];
         foreach ($productAttributes as $attribute) {
```
Comment (BarnyShergold) [inline app/code/Magento/CatalogWidget/Model/Rule/Condition/Product.php:106]: As the method loadAttributeOptions() is being functionally changed, the testsuite test for the method should be updated to take into consideration the attribute can now affect the output.

**Review by BarnyShergold — CHANGES_REQUESTED** [review]

**Comment by ishakhsuvarov** [comment]
Hi @evgeniykapelko 

I am closing this PR for now as it seems like it has become stale and still needs work.

Please feel free to reopen if you wish to continue.

Thanks!

**Pull request closed without merging**